### DISCUSSION TITLE
API Call data collection - Dynamic Variables

### DISCUSSION THREAD

**Discussion by dauphinpasdroit** [Budibase/budibase#11335] [Help]
Hello there !

I'm having some issue to deal with the dynamic variables with the API Call data collection features.

My use case is this one :

- Do a first Azure API call to get a bearer token.
- Use this bearer token into another API call to get Virtual Machine name in a specific resource group or subscriptin, whatever.

The transformer part is setup like this: 

![image](https://github.com/Budibase/budibase/assets/110522034/311aace0-84a6-4dea-9647-0691c679c8c0)

which return the value, like this in json format:

![image](https://github.com/Budibase/budibase/assets/110522034/13fc4a05-5b0b-4bac-91bf-353813a7cbdb)

I create a dynamic variable here: 

![image](https://github.com/Budibase/budibase/assets/110522034/b9f5a5dc-c5d9-401e-90bb-6d9f8fb72db3)

and the previous looks good : 

![image](https://github.com/Budibase/budibase/assets/110522034/4f9fa990-5a0f-4b9d-99af-0627f8799a5f)

![image](https://github.com/Budibase/budibase/assets/110522034/ae0c11a9-3734-41de-8337-95842c428eaa)

Then, I setup my second API call, with the dynamic value I want to use as bearer token in an Authorization headers:

![image](https://github.com/Budibase/budibase/assets/110522034/bd75608e-315a-4382-af82-8ce8a1a1c37d)

But it keeps on failing and return: 

![image](https://github.com/Budibase/budibase/assets/110522034/1f500206-3f45-4a42-897d-fc60dcf853c3)

My questions are:

- Is there a way to print out the dynamic variable somewhere to check it ?
- Is there a way to add it into a database ?
- Is there any charactere limitation regarding a value of a dynamic variable ?

Thank you for your help :)

**Comment by melohagan** [comment]
Hey @alphaxr6 

Is it simply that you need to remove the double quotes from the binding reference. I.e. `Bearer {{ Dynamic.access_token }}` ?

---

> Is there a way to print out the dynamic variable somewhere to check it

You could try using a JavaScript binding instead, that way you could use a `console.log` statement

> Is there any character limitation regarding a value of a dynamic variable ?

 Don't think so!
